src/stores/zkopru.js

```diff
@@ -217,6 +221,75 @@ export default {
           from: rootState.account.accounts[0],
         }]
       })
+    },
+    loadHistory: async ({ state, rootState }) => {
+      const { layer2, db } = state.client.node
+      const { web3 } = state.client.node.layer1
+      const l2Address = state.wallet.wallet.account.zkAddress.toString()
+      const l1Address = rootState.account.accounts[0]
+      const deposits = await db.findMany('Deposit', {
+        where: {
+          ownerAddress: [l2Address],
+        }
+      }) 
+
+      // because case sensitivity differes in l1Address and database,
+      // need to filter after querying all records in database.
+      const withdrawals = (await db.findMany('Withdrawal', { where: {}, include: { proposal: true } }))
+        .filter(withdraw => withdraw.to.toLocaleLowerCase() === l1Address)
+      const sendTxs = await db.findMany('Tx', {
+        where: { senderAddress: l2Address },
+        include: { proposal: true }
+      })
+      const receiveTxs = await db.findMany('Tx', {
+        where: { receiverAddress: l2Address },
+        include: { proposal: true } 
+      })
+      const pendingTxs = await db.findMany('PendingTx', {
+        where: { senderAddress: l2Address },
+      })
+
+      // use utxos to get deposit amount
+      const utxos = await db.findMany('Utxo', {
+        where: {
+          owner: [l2Address],
+        }
+      })
+
+      const history = [
+        ...await Promise.all(deposits.map(async (deposit) => {
+          const utxo = utxos.find(utxo => utxo.hash === deposit.note)
+          const block = await web3.eth.getBlock(deposit.blockNumber)
```

Comment:
Ideally we'd get rid of this web3 request. The problem is the proposal doesn't exist in the db until after a confirmation period.

I think I'll make a change so that the deposit looks like this:
```
{
  ...depositFields,
  proposal?: data, // exists if the block is confirmed
  pendingProposal?: data // exists if there's a pending proposal
}
```

This change will need to be made in the Zkopru library itself, I'll let you know what I come up with. The pending system needs to be much more robust.

Reply:
ok, I agree.